pkg/virt-handler/vm.go

```diff
@@ -566,6 +566,13 @@ func (d *VirtualMachineController) setupNetwork(vmi *v1.VirtualMachineInstance,
 				return neterrors.CreateCriticalNetworkError(fmt.Errorf("failed to set up tun device, %s", err))
 			}
 		}
+		vdpaDevs, _ := filepath.Glob(fmt.Sprintf("/proc/%d/root/dev/vhost-vdpa-*", isolationRes.Pid()))
```

Comment:
Sorry, I don't know the code well, but I'm wondering, here how do you know it's a virtio-net device?

/dev/vhost-vdpa-* can be any virtio device emulated via vDPA.

Here is an example of 2 devices (block and net):
```bash
$ ls -l /dev/vhost-vdpa-*
crw-------. 1 root root 511, 0 Jul 14 10:55 /dev/vhost-vdpa-0
crw-------. 1 root root 511, 1 Jul 14 10:55 /dev/vhost-vdpa-1

$ vdpa dev
vdpa42: type block mgmtdev vdpasim_blk vendor_id 0 max_vqs 1 max_vq_size 256
vdpa13: type network mgmtdev vdpasim_net vendor_id 0 max_vqs 3 max_vq_size 256

$ ls /sys/devices/vdpa42/ | grep vhost
vhost-vdpa-0

$ cat /sys/devices/vdpa42/vhost-vdpa-*/dev
511:0

$ ls /sys/devices/vdpa13/ | grep vhost
vhost-vdpa-1

$ cat /sys/devices/vdpa13/vhost-vdpa-*/dev
511:1
```

Reply:
@stefano-garzarella The vDPA devices in virt-launcher is allocated by device plugin, and there should be a configuration file that specifies the device type(1000). Here's my configuration file for the [Intel SR-IOV device plugin](https://github.com/intel/sriov-network-device-plugin):
```
# kubectl get cm -n kube-system sriovdp-config -o yaml
apiVersion: v1
data:
  config.json: |
    {
        "resourceList": [
            {
                "resourceName": "vdpa_jaguar_vhost",
                "selectors": [{
                    "vendors": ["1f53"],
                    "devices": ["1000"],
                    "drivers": ["jaguar"],
                    "vdpaType": "vhost"
                }]
            }
        ]
    }
```

Reply:
@stefano-garzarella The line of code you mentioned looks for the vDPA device assigned by the device plugin and chown to qemu:qemu. If you don't do this, libvirt will tell you that you don't have enough privileges.

Reply:
@makoto126 sorry, I don't know the code very well, but my concern is more about how to know if it's really the device we want. If we also add support for block devices in the future, should we change this code?

Reply:
@stefano-garzarella If vDPA block support is added in the future, block devices also need to be chown to qemu:qemu, it shouldn't matter too much. If we must know whether the device is what we want, can read from /proc/%d/environ to get the same env variables with virt-launcher and parse the content, i don't think it's really necessary.

Reply:
@makoto126 got it, thanks!

Reply:
@makoto126 aren't you changing the permission of all devices `dev/vhost-vdpa-*` on the node in this way? What if there are vhost-vdpa devices that don't need to be assigned and used by KubeVirt?

Reply:
@makoto126 another question: are we changing the permission of all `vhost-vdpa-*` devices on the node? If it is the case, is that safe?

Reply:
@alicefr @stefano-garzarella No, only change the permission of all vhost-vdpa-* devices that assigned by the device plugin. 
For example, if the node has 5 vdpa devices, and the vmi declares 2 vdpa devices：
```
          interfaces:
          - masquerade: {}
            name: default
          - name: offload
            vdpa: {}
          - name: offload2
            vdpa: {}
      networks:
      - name: default
        pod: {}
      - multus:
          networkName: default/offload
        name: offload
      - multus:
          networkName: default/offload
        name: offload2
```
There will only be 2 vdpa devices in the virt-launcher container.

Reply:
ah I see, you are using the container pid, thanks for the explanation